### PR TITLE
fix(circleci): add api route prefix to proxy rewrite path

### PR DESCRIPTION
**Pull request by andrewthauer**
## Hey, I just made a Pull Request!

Looks like this is also a required change for the circleci plugin to work with the recent backend `/api` route changes.

#### :heavy_check_mark: Checklist

- [ ] All tests are passing `yarn test`
- [ ] Screenshots attached (for UI changes)
- [ ] Relevant documentation updated
- [ ] Prettier run on changed files
- [ ] Tests added for new functionality
- [ ] Regression tests added for bug fixes

### CODE REVIEW

**Review by Rugvip — APPROVED** [review]
Oh wut? Hmm, kinda feeling like the `/api/proxy` prefix should always be rewritten no matter what.

**Comment by andrewthauer** [comment]
> Oh wut? Hmm, kinda feeling like the `/api/proxy` prefix should always be rewritten no matter what.

Yeah, maybe the plugin should do that automatically to avoid having to do it in the config?

File: app-config.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ proxy:
     target: https://circleci.com/api/v1.1
     changeOrigin: true
     pathRewrite:
-      '^/proxy/circleci/api/': '/'
+      '^/api/proxy/circleci/api/': '/'
```
Comment (timja) [inline app-config.yaml:21]: is this even needed? 

isn't it the default? the Jenkins plugin works without it

Comment (andrewthauer) [reply app-config.yaml:21]: Apparently 🤷. I just confirmed again on master it only works by adding the `/api`.

Comment (timja) [reply app-config.yaml:21]: i meant removing path rewrite, is that what you tried?

Comment (andrewthauer) [reply app-config.yaml:21]: Yeah, your completely right. If I remove the `pathRewrite` and `changeOrigin` works fine. Thanks.

**Comment by andrewthauer** [comment]
I've removed the `pathRewrite` as @timja pointed out it's not actually required. Also updated the relevant docs to reflect this change along with the the new `/api/proxy` routing prefix.

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**